commands/audit/sca/swift/swift.go

```diff
@@ -0,0 +1,221 @@
+		foundDependency = false
+	}
+	return foundDependency, tempIndex, startLine, startCol
+}
+
+func FixTechDependency(dependencyName, dependencyVersion, fixVersion string, descriptorPaths ...string) error {
+	for _, descriptorPath := range descriptorPaths {
+		path.Clean(descriptorPath)
+		if !strings.HasSuffix(descriptorPath, "Package.swift") {
+			log.Logger.Warn("Cannot support other files besides Package.swift: %s", descriptorPath)
+			continue
+		}
+		data, err := os.ReadFile(descriptorPath)
+		var newLines []string
+		if err != nil {
+			continue
+		}
+		lines := strings.Split(string(data), "\n")
+		foundDependency := false
+		var tempIndex int
+		for index, line := range lines {
+			if strings.Contains(line, dependencyName) {
+				foundDependency = true
+				tempIndex = index
+			}
+			// This means we are in a new dependency (we cannot find dependency name and version together)
+			//nolint:gocritic
+			if index > tempIndex && foundDependency && strings.Contains(line, ".package") {
+				foundDependency = false
+			} else if foundDependency && strings.Contains(line, dependencyVersion) {
+				newLine := strings.Replace(line, dependencyVersion, fixVersion, 1)
+				newLines = append(newLines, newLine)
+				foundDependency = false
+			} else {
+				newLines = append(newLines, line)
+			}
+		}
+		output := strings.Join(newLines, "\n")
+		err = os.WriteFile(descriptorPath, []byte(output), 0644)
+		if err != nil {
+			return fmt.Errorf("failed to write file: %v", err)
+		}
+	}
+	return nil
+}
+
+func GetSwiftDependenciesGraph(data *Dependencies, dependencyMap map[string][]string, versionMap map[string]string) {
+	data.Name = strings.TrimSuffix(data.Name, ".git")
+	data.Name = strings.TrimPrefix(data.Name, "https://")
+	_, ok := dependencyMap[data.Name]
+	if !ok {
+		dependencyMap[data.Name] = []string{}
+		versionMap[data.Name] = data.Version
+	}
+	for _, dependency := range data.Dependencies {
+		dependency.Name = strings.TrimSuffix(dependency.Name, ".git")
+		dependency.Name = strings.TrimPrefix(dependency.Name, "https://")
+		dependencyMap[data.Name] = append(dependencyMap[data.Name], dependency.Name)
+		GetSwiftDependenciesGraph(dependency, dependencyMap, versionMap)
+	}
+}
+
+func GetDependenciesData(exePath, currentDir string) (*Dependencies, error) {
+	result, _, err := runSwiftCmd(exePath, currentDir, []string{"package", "show-dependencies", "--format", "json"})
+	if err != nil {
+		return nil, err
+	}
+	var data *Dependencies
+	err = json.Unmarshal(result, &data)
+	if err != nil {
+		return nil, err
+	}
+	return data, nil
+}
+
+func BuildDependencyTree(params utils.AuditParams) (dependencyTree []*xrayUtils.GraphNode, uniqueDeps []string, err error) {
+	currentDir, err := coreutils.GetWorkingDirectory()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	packageName := filepath.Base(currentDir)
+	packageInfo := fmt.Sprintf("%s:%s", packageName, VersionForMainModule)
```

Comment:
Technically I can do that, but since its a code file (Package.swift) and not a json or xml or something with something I can unmarshal I would have to read the file and take the line below Package(, which I'm not sure is "good" coding. But to be honest why do I care about the package name? It really does not appear anywhere on the results